### PR TITLE
Wizard recipe: SDL2_image-v2.0.5

### PR DESCRIPTION
**Pull request by aviks**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: SDL2_image
* Version: v2.0.5

@staticfloat please review and merge.

### CODE REVIEW

**Comment by aviks** [comment]
cc: @giordano 

This fails on windows with the following error. The windows dll is in `bin` and is called `SDL2_image.dll`, not `libSDL2_image.dll`

```
┌ Warning: Could not find correspondences for libSDL2_image
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/8ST8p/src/wizard/utils.jl:128
ERROR: Some build products could not be found (see above).
```

Separately, this cannot find libtiff or libjpeg, even though they are in the dependencies. However, it does find libpng and libwebp, so this can be considered good to go for now on that front.

File: S/SDL2_image/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    Linux(:x86_64, libc=:musl),
+    Linux(:aarch64, libc=:musl),
+    Linux(:armv7l, libc=:musl, call_abi=:eabihf),
+    MacOS(:x86_64)
+]
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libSDL2_image", :libsdl2_image)
```
Comment (giordano) [inline S/SDL2_image/build_tarballs.jl]: For the Windows issue:
```suggestion
    LibraryProduct(["libSDL2_image", "SDL2_image"], :libsdl2_image)
```
In `platforms` just do
```julia
platforms = supported_platforms()
```

**Comment by aviks** [comment]
Only FreeBSD now fails, with a similar. 

```
[ Info: Could not locate libSDL2_image, SDL2_image inside ["/workspace/S/SDL2_image/build/x86_64-unknown-freebsd11.1/JlGGEGA7/destdir/lib64", "/workspace/S/SDL2_image/build/x86_64-unknown-freebsd11.1/JlGGEGA7/destdir/lib"]
┌ Error: Built SDL2_image but libsdl2_image still unsatisfied:
└ @ BinaryBuilder ~/.julia/dev/BinaryBuilder/src/AutoBuild.jl:753
ERROR: LoadError: Cannot continue with unsatisfied build products!
```

**Comment by aviks** [comment]
Woohoo! All builds are green! Thanks Mose!

**Comment by aviks** [comment]
This build now includes support for PNG, TIFF, WEBP and JPEG.

**Comment by giordano** [comment]
> This build now includes support for PNG, TIFF, WEBP and JPEG.

I'd explicitly specify the libraries for PNG, TIFF, JPEG, even if they're installed automatically by `libwebp_jll`

**Pull request merged**